### ISSUE TITLE
No borders in popupmenus

### ISSUE THREAD

**Issue by CDrummond** [MartinBriza/adwaita-qt#90]
Latest adwaita-qt has removed borders from pop-up menus. However, there should be a light border the same shade grey as the menu separator.

In Fedora 25 and Ubuntu GNOME 16.10 Adwaita-Gtk has a border around menus.

**Comment by MartinBriza** [comment]
Same as in the other bug: Hmmm, I may have messed something up when tweaking menu borders! Thanks for noticing, once again!

**Comment by CDrummond** [comment]
The incorrect change is https://github.com/MartinBriza/adwaita-qt/commit/d887d569083206f8ce1351359cf5988d72152a0d this was to fix #81 

However, #81 just stated the borders were too dark (which they were). Your change just removes the border, when all it needed to do was alter the shade.

**Comment by MartinBriza** [comment]
I suspect the menu border in GTK apps is caused by the shadow which is rendered by the window manager. It's pretty hard to be sure about this though, however the borders I see in Qt apps is now very similar to what I see in gtk3-widget-factory...

**Comment by MartinBriza** [comment]
The only big difference I see is that the empty gaps at the top and bottom should be just 2px instead of 3.

**Comment by CDrummond** [comment]
Perhaps this is another Gtk-version difference. But, in Ubuntu 16.10 with Gtk 3.20 there is most definitely a 1pixel non-shadow border (of the same shade) around the whole popup menu.

With Adwaita-Qt there is now no border, and **only** the shadow. This is most obvious at the top of a context menu on a white background - the top seems to vanish.

**Comment by MartinBriza** [comment]
I can't take a screenshot of what it looks like on my system (GNOME 3.24.0), however there's no border, only the 1px gray transparent shadow all around the menus (even context menus).

**Comment by MartinBriza** [comment]
Ha, found a way how to take a screenshot of it. This is the GTK3 Widget Factory.
![untitled](https://cloud.githubusercontent.com/assets/3977440/24862996/652ca22a-1dff-11e7-9254-2e78cb2f931e.png)

**Comment by MartinBriza** [comment]
And this is the context menu in KWrite
![untitled](https://cloud.githubusercontent.com/assets/3977440/24863102/b90438fe-1dff-11e7-83f9-af2ec375fc60.png)

**Comment by CDrummond** [comment]
Yup, your's look great. I guess this could be a gnome shell version difference. Zooming in on your gtk3 example the grey border is clearly a shadow (as in in drawn by the window manager, as it is semi-transparent). I guess you should just close this then. Hopefully gnome shell in 17.04 will be better :-)

**Comment by MartinBriza** [comment]
> Before the change to remove the border - did you have a 2 pixel border instead? If not, then could you not re-add the code and use a semi-transparent border?

I don't understand, sorry.

**Comment by CDrummond** [comment]
Previously, my popup menus had 1 pixel border. There was a bug report that this border was too dark - your previous change was to fix this, but it also removed the border I was seeing.

So, seeing as you still have a border, and I do not, I was wondering if before you made the change to remove the border, if you have the border/shadow shown in your screenshots AND the dark border that you removed. If so, then this border would have been 2 pixels wide - one for the dark border, and 1 for the shadow/border.

If it was not 2 pixels wide - then that would imply the dark border was drawn **over** the shadow border. If this was the case, then surely we could just draw a light grey border instead - that way you;d see a similar effect to what you are seeing (although not semi-transparent), and I'd still have a border.

**Comment by MartinBriza** [comment]
Oh, I see. Well, no, the actual border was just 1px. I just tried checking out the commit before I removed the border and there's just a very dark border right inside the menu "window" + one pixel of the shadow which is still the same, regardless of the theme code.
I'm sorry, I don't think there's a solution for us both. Same as with a similar conflict in the past, I'd just stick with what seems to be the newer way of doing things in upstream Adwaita.
